Assets/MixedRealityToolkit/Definitions/Utilities/ArticulatedHandPose.cs

```diff
@@ -0,0 +1,286 @@
+        {
+            for (int i = 0; i < jointCount; i++)
+            {
+                var p = Vector3.Lerp(poseA.localJointPoses[i].Position, poseB.localJointPoses[i].Position, value);
+                var r = Quaternion.Slerp(poseA.localJointPoses[i].Rotation, poseB.localJointPoses[i].Rotation, value);
+                localJointPoses[i] = new MixedRealityPose(p, r);
+            }
+        }
+
+        public void TransitionTo(ArticulatedHandPose other, float lastAnim, float currentAnim)
+        {
+            if (currentAnim <= lastAnim)
+            {
+                return;
+            }
+
+            float range = Mathf.Clamp01(1.0f - lastAnim);
+            float lerpFactor = range > 0.0f ? (currentAnim - lastAnim) / range : 1.0f;
+            for (int i = 0; i < jointCount; i++)
+            {
+                var p = Vector3.Lerp(localJointPoses[i].Position, other.localJointPoses[i].Position, lerpFactor);
+                var r = Quaternion.Slerp(localJointPoses[i].Rotation, other.localJointPoses[i].Rotation, lerpFactor);
+                localJointPoses[i] = new MixedRealityPose(p, r);
+            }
+        }
+
+        public enum GestureId
+        {
+            /// <summary>
+            /// Unspecified hand shape
+            /// </summary>
+            None = 0,
+            /// <summary>
+            /// Flat hand with fingers spread out
+            /// </summary>
+            Flat,
+            /// <summary>
+            /// Relaxed hand pose
+            /// </summary>
+            Open,
+            /// <summary>
+            /// Index finger and Thumb touching, index tip does not move
+            /// </summary>
+            Pinch,
+            /// <summary>
+            /// Index finger and Thumb touching, wrist does not move
+            /// </summary>
+            PinchSteadyWrist,
+            /// <summary>
+            /// Index finger stretched out
+            /// </summary>
+            Poke,
+            /// <summary>
+            /// Grab with whole hand, fist shape
+            /// </summary>
+            Grab,
+            /// <summary>
+            /// OK sign
+            /// </summary>
+            ThumbsUp,
+            /// <summary>
+            /// Victory sign
+            /// </summary>
+            Victory,
+        }
+
+        private static readonly Dictionary<GestureId, ArticulatedHandPose> handPoses = new Dictionary<GestureId, ArticulatedHandPose>();
+
+        public static ArticulatedHandPose GetGesturePose(GestureId gesture)
+        {
+            if (handPoses.TryGetValue(gesture, out ArticulatedHandPose pose))
+            {
+                return pose;
+            }
+            return null;
+        }
+
+        #if UNITY_EDITOR
+        public static void LoadGesturePoses()
+        {
+            string basePath = "Assets/MixedRealityToolkit.Services/InputSimulation/ArticulatedHandPoses/";
```

Comment:
Do we have any better way of addressing file paths?

Reply:
I can't find anything better looking for "Assets/"

---
In reply to: [281623574](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4211#discussion_r281623574) [](ancestors = 281623574)

Reply:
This will cause additional issues for https://github.com/microsoft/MixedRealityToolkit-Unity/issues/4067.

@wiwei mentions in that issue having done some relative file path work. It'd be good if we can leverage that here as well.